PWGCF/DataModel/CorrelationsDerived.h

```diff
@@ -79,6 +79,48 @@ using CFTrack = CFTracks::iterator;
 using CFTrackLabel = CFTrackLabels::iterator;
 using CFTracksWithLabel = soa::Join<CFTracks, CFTrackLabels>;
 using CFTrackWithLabel = CFTracksWithLabel::iterator;
+
+//------transient CF-filter to CF-2prong-filter
+namespace cfcollref
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+} // namespace cfcollref
+DECLARE_SOA_TABLE(CFCollRefs, "AOD", "CFCOLLREF", o2::soa::Index<>, cfcollref::CollisionId); //! Transient cf collision index table
+
+using CFCollRef = CFCollRefs::iterator;
+
+namespace cftrackref
+{
+DECLARE_SOA_INDEX_COLUMN(Collision, collision);
+DECLARE_SOA_INDEX_COLUMN(Track, track);
+// DECLARE_SOA_INDEX_COLUMN(CFTrack, cfTrack); //! CF track index
+} // namespace cftrackref
+DECLARE_SOA_TABLE(CFTrackRefs, "AOD", "CFTRACKREF", o2::soa::Index<>, cftrackref::CollisionId, cftrackref::TrackId); //! Transient cf track index table
+// DECLARE_SOA_TABLE(CFTrackRefs, "AOD", "CFTRACKREF", track::CollisionId); //! Transient cf track index table
+
+using CFTrackRef = CFTrackRefs::iterator;
+//------
+
+namespace cf2prongtrack
+{
+DECLARE_SOA_INDEX_COLUMN(CFCollision, cfCollision);                                                                                                     //! Index to collision
+DECLARE_SOA_INDEX_COLUMN(CFMcParticle, cfMCParticle);                                                                                                   //! Index to MC particle
+DECLARE_SOA_INDEX_COLUMN_FULL(CFTrackProng0, cfTrackProng0, std::result_of<decltype (&aod::CFTrack::globalIndex)(aod::CFTrack)>::type, CFTracks, "_0"); //! Index to prong 1 track
+DECLARE_SOA_INDEX_COLUMN_FULL(CFTrackProng1, cfTrackProng1, std::result_of<decltype (&aod::CFTrack::globalIndex)(aod::CFTrack)>::type, CFTracks, "_1"); //! Index to prong 1 track
```

Comment:
```suggestion
DECLARE_SOA_INDEX_COLUMN_FULL(CFTrackProng0, cfTrackProng0, int, CFTracks, "_0"); //! Index to prong 1 track
DECLARE_SOA_INDEX_COLUMN_FULL(CFTrackProng1, cfTrackProng1, int, CFTracks, "_1"); //! Index to prong 1 track
```

Reply:
Isn't global index int64 though? Just int might not be enough?

Reply:
indices in O2 are all int, you have no choice :)

And to exceed int32, you need to put more than 4 GB of entries in a file which is basically impossible for a TF size of a few 100 MBs...